src/main/java/HeadersRestTemplateService.java

```diff
@@ -0,0 +1,28 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Service;
+
+@Service
+public class HeadersRestTemplateService {
+
+    @Value("${github.token}")
+    private String githubToken;
+
+    @Value("${openai.api.key}")
+    private String openaiApiKey;
+
+    public HttpHeaders createGitHubHeaders() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + githubToken);
+        headers.set("Content-Type", "application/json");
+        return headers;
+    }
+
+    public HttpHeaders createOpenAiHeaders() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + openaiApiKey);
+        headers.set("Content-Type", "application/json");
+        return headers;
+    }
+
+}
```

Comment:
The class is responsible for creating headers for different services, which violates the Single Responsibility Principle (SRP). It could be refactored to separate concerns, such as creating a dedicated class for each service's headers. (Severity: Medium)
**Suggestion**: Consider creating separate classes for GitHub and OpenAI headers, or a common interface if they share functionality.